Test/unicodechars/dafny0/Char.dfy

```diff
@@ -0,0 +1,117 @@
+// RUN: %dafny_0 /compile:0 /unicodeChar:1 /print:"%t.print" /dprint:"%t.dprint" "%s" > "%t"
```

Comment:
Should this file (and possibly some of the others) be in `Test/comp` instead? Or, if there are many of them, I suppose they could also go into a new directory like `Test/unicode`. (These are points to consider, not hard suggestions from me.)

Reply:
I intended `Test/unicodechars` to imitate the `allocated1` directory that includes several test files from elsewhere under `Test` and reproduces the folder layout to a degree. For some, like this one, I couldn't just include the original because it uses `\u` escapes that are forbidden in `--unicode-char` mode, and I needed to add more test cases specific to that mode.

The real question might be whether I should still consider the main `UnicodeStrings.dfy` test file the `--unicode-char` version of `Test/dafny0/Strings.dfy, and therefore move it to `Test/unicodechars/dafny0/Strings.dfy`. It's obviously fairly different now, but in spirit I think that makes sense.